### PR TITLE
Improve matcher: add effect to array if no allow or deny is provided

### PR DESCRIPTION
**Pull request by ErikUniformAgri**
- Improve matcher: add effect to array if no allow or deny is provided
- Don't use superuser in console (conflicts with the internal superuser/admin)
- Fix KeyMatchDeny which somehow now works the same way it does in go

### CODE REVIEW

**Comment by hsluoyz** [comment]
@jkour plz review

**Comment by jkour** [comment]
Hi guys,

I will check in the next couple of days. 

Thanks

File: SourceCode/Common/Casbin/Casbin.pas [hunk 1]
```diff
@@ -309,18 +309,20 @@ function TCasbin.enforce(const aParams: TEnforceParameters; const aPointer:
...
-              matcherResult:=erAllow
+            begin
+              matcherResult:=erAllow;
+            end
             else
             begin
               var f: string :=  policyList[request.Count];
               if policyList.Count > request.Count then
-                if UpperCase(policyList[request.Count]) = 'ALLOW' then
+                if SameText(Trim(policyList[request.Count]), 'ALLOW') then
```
Comment (jkour) [inline SourceCode/Common/Casbin/Casbin.pas:319]: I used the TStringHelper to achieve the same here.

File: SourceCode/Common/Casbin/Casbin.pas [hunk 2]
```diff
@@ -309,18 +309,20 @@ function TCasbin.enforce(const aParams: TEnforceParameters; const aPointer:
...
                   matcherResult:=erAllow
                 else
                   matcherResult:=erDeny;
-              SetLength(effectArray, Length(effectArray)+1);
-              effectArray[Length(effectArray)-1]:=matcherResult; //PALOFF
             end;
+            SetLength(effectArray, Length(effectArray)+1);
+            effectArray[Length(effectArray)-1]:=matcherResult; //PALOFF
```
Comment (jkour) [inline SourceCode/Common/Casbin/Casbin.pas:325]: Ah yes...Thanks for spotting it

**Comment by jkour** [comment]
Done. Thanks

**Pull request merged**